src/Connection.php

```diff
@@ -104,43 +128,19 @@ public static function normalizeDSN($dsn, $user = null, $pass = null)
      *
      * @return Connection
      */
-    public static function connect($dsn, $user = null, $password = null, $args = [])
+    public static function establish($dsn, $user = null, $password = null, $args = [])
     {
         // If it's already PDO object, then we simply use it
         if ($dsn instanceof \PDO) {
             $driverType = $dsn->getAttribute(\PDO::ATTR_DRIVER_NAME);
-            $connectionClass = self::class;
-            $queryClass = null;
-            $expressionClass = null;
-            switch ($driverType) {
-                case 'pgsql':
-                    $connectionClass = Postgresql\Connection::class;
-                    $queryClass = Postgresql\Query::class;
-
-                    break;
-                case 'oci':
-                    $connectionClass = Oracle\Connection::class;
-
-                    break;
-                case 'sqlite':
-                    $queryClass = Sqlite\Query::class;
-
-                    break;
-                case 'mysql':
-                    $expressionClass = Mysql\Expression::class;
-                    // no break
-                default:
-                    // Default, for backwards compatibility
-                    $queryClass = Mysql\Query::class;
-
-                    break;
-            }
+
+            /**
+             * @var Connection $driverClass
+             */
+            $connectionClass = self::resolveDriver($driverType);
```

Comment:
It does not resolve driver!

It does resolve connection class!

Driver is PDO, nothing else. There should be no driver elsewhere.

Reply:
You need to dig a bit deeper in DSQL first.

I spent a lot of time last PR to figure out how to call the peoperties and classes as Connections are nested.
So it is connecton -> connection -> connecton -> PDO or connection or whatever that has `execute` method)

So finally now I decided to keep the old naming just call the wrapped connection "driverConnection".
`resolveDriver` gets a driver alias and resolves it to a connection class

Reply:
I know, it is driver if not nested in execute.

But the new design should focus on proving strictly connection (you can still wrap it ;-))

Reply:
Yes, we can think of further redesign but this PRs is targeted at removing direct coupling of drivers in code and introducing the extendable registry

Reply:
As long as you have this annotated to return an Connection (type should be added to the code anyway), than this name is 100% wrong. The same for all registry related names.